src/main/groovy/com/github/eirnym/js2p/JsonSchemaPlugin.groovy

```diff
@@ -30,9 +30,15 @@ class JsonSchemaPlugin implements Plugin<Project> {
     @Override
     public void apply(Project project) {
         project.extensions.create('jsonSchema2Pojo', JsonSchemaExtension)
+        project.plugins.withId('java') {
+            def js2pTask = project.tasks.register('generateJsonSchema2Pojo', GenerateJsonSchemaJavaTask)
+            project.tasks.named('compileJava').configure {
+                it.dependsOn(js2pTask)
+            }
+        }
         project.afterEvaluate {
             if (project.plugins.hasPlugin('java')) {
-                project.tasks.create('generateJsonSchema2Pojo', GenerateJsonSchemaJavaTask)
+                // do nothing
```

Comment:
do we need this `if` branch?

Reply:
It's possible to remove like below:

```patch
diff --git a/src/main/groovy/com/github/eirnym/js2p/JsonSchemaPlugin.groovy b/src/main/groovy/com/github/eirnym/js2p/JsonSchemaPlugin.groovy
index 3435303..ad07b68 100644
--- a/src/main/groovy/com/github/eirnym/js2p/JsonSchemaPlugin.groovy
+++ b/src/main/groovy/com/github/eirnym/js2p/JsonSchemaPlugin.groovy
@@ -37,9 +37,7 @@ class JsonSchemaPlugin implements Plugin<Project> {
             }
         }
         project.afterEvaluate {
-            if (project.plugins.hasPlugin('java')) {
-                // do nothing
-            } else if (project.plugins.hasPlugin('com.android.application') || project.plugins.hasPlugin('com.android.library')) {
+            if (project.plugins.hasPlugin('com.android.application') || project.plugins.hasPlugin('com.android.library')) {
                 def config = project.jsonSchema2Pojo
                 def variants = null
                 if (project.android.hasProperty('applicationVariants')) {
@@ -59,7 +57,7 @@ class JsonSchemaPlugin implements Plugin<Project> {
 
                     variant.registerJavaGeneratingTask(task, (File) task.outputDir)
                 }
-            } else {
+            } else if (! project.plugins.hasPlugin('java')) {
                 for (Plugin<?> plugin : project.plugins) {
                     project.logger.error(plugin.class.name);
                 }
```

I felt that the last `if (! project.plugins.hasPlugin('java'))` isn't so intuitive.
It's why I keep the `if` branch for java plugin.

Reply:
it's fine by me :)